docs/sources/installation/mac.md

```diff
@@ -1,116 +1,315 @@
+
+		CONTAINER ID        IMAGE               COMMAND                CREATED             STATUS              PORTS                                           NAMES
+		5fb65ff765e9        nginx:latest        "nginx -g 'daemon of   3 minutes ago       Up 3 minutes        0.0.0.0:49156->443/tcp, 0.0.0.0:49157->80/tcp   web  
+
+	At this point, you can see `nginx` is running as a daemon. 
+
+3. View just the container's ports.
+
+		$ docker port web
+		443/tcp -> 0.0.0.0:49156
+		80/tcp -> 0.0.0.0:49157
+	
+	This tells you that the `web` container's port `80` is mapped to port
+	`49157` on your Docker host.
+
+4. Enter the `https://localhost:49157` address (`localhost` is `0.0.0.0`) in your browser:
+
+	   ![Bad Address](/installation/images/bad_host.png)
+
+	This didn't work. The reason it doesn't work is your `DOCKER_HOST` address is
+	not the localhost address (0.0.0.0) but is instead the address of the
+	`boot2docker` VM.
+
+5. Get the address of the `boot2docker` VM.
+
+		$ boot2docker ip
+		192.168.59.103
+		
+6. Enter the `https://192.168.59.103:49157` address in your browser:
+
+	![Correct Addressing](/installation/images/good_host.png)
+
+	Success!
+
+7. To stop and then remove your running `nginx` container, do the following:
+
+		$ docker stop web
+		$ docker rm web
+		
+### Mount a volume on the container
+
+When you start `boot2docker`, it automatically shares your `/Users` directory
+with the VM. You can use this share point to mount directories onto your container.
+The next exercise demonstrates how to do this.
+
+1. Change to your user `$HOME` directory.
+
+		$ cd $HOME
+	
+2. Make a new `site` directory.
+
+		$ mkdir site
+
+3. Change into the `site` directory.
+
+		$ cd site
+
+4. Create a new `index.html` file.
+
+		$ echo "my new site" > index.html
+	
+5. Start a new `nginx` container and replace the `html` folder with your `site` directory.
+
+		$ docker run -d -P -v /Users/mary/site:/usr/share/nginx/html --name mysite nginx
+	
+6. Get the `mysite` container's port.
+
+		$ docker port mysite
+		80/tcp -> 0.0.0.0:49166
+		443/tcp -> 0.0.0.0:49165
+	
+7. Open the site in a browser:
+
+	![My site page](/installation/images/newsite_view.png)
+
+8. Try adding a page to your `$HOME/site` in real time.
+
+		$ echo "This is cool" > cool.html
+	
+9. Open the new page in the browser.
+
+	![Cool page](/installation/images/cool_view.png)
+
+9. Stop and then remove your running `mysite` container.
+
+		$ docker stop mysite
+		$ docker rm mysite
+
+## Upgrade Boot2Docker 
+
+If you running Boot2Docker 1.4.1 or greater, you can upgrade Boot2Docker from
+the command line. If you are running an older version, you should use the
+package provided by the `boot2docker` repository.
+
+### From the command line
+
+To upgrade from 1.4.1 or greater, you can do this:
+
+1. Open a terminal on your local machine.
+
+2. Stop the `boot2docker` application.
+
+		$ boot2docker stop
+			
+3. Run the upgrade command.
+
+		$ boot2docker upgrade
```

Comment:
mmm, probably worth telling them they then need to  run `boot2docker up` and shellinit again, or close the Boot2Docker app shell and start a new one